utils/getClaimStatus.tsx

```diff
@@ -7,66 +7,115 @@ import { ClaimStatusContent, I18nString, TextOptionalLink, TransLineProps } from
-  const linkKeys = summary.links
+/**
+ * Build props to pass to the TransLine react component.
+ */
+export function buildTransLineProps(json: TextOptionalLink, i18nKey: I18nString): TransLineProps {
+  return {
+    i18nKey: i18nKey,
+    links: buildTransLineLinks(json.links),
+  }
+}
 
-  // Lookup each link and send the array to the TransLine component.
-  if (linkKeys && linkKeys.length > 0) {
-    for (const linkKey of linkKeys) {
-      // Explicitly cast to one of the allowed keys in urls.json
-      const key = linkKey as UrlType
-      const url = getUrl(key)
-      if (url) {
-        transLineProps.links.push(url)
-      }
-    }
+/**
+ * Helper function to construct the i18nKey for claim status link text.
+ */
+function buildI18nKey(keys: string[]): I18nString {
+  return 'claim-status:' + keys.join('.') + '.text'
+}
+
+/**
+ * Get Claim Status summary.
+ */
+export function buildClaimStatusSummary(
+  scenarioObject: ClaimStatusScenarioJson,
+  scenarioString: string,
+): TransLineProps {
+  const keys = ['scenarios', scenarioString, 'summary']
+  return buildTransLineProps(scenarioObject.summary, buildI18nKey(keys))
+}
+
+/**
+ * Get next steps content for Claim Status.
+ */
+export function buildNextSteps(
+  scenarioObject: ClaimStatusScenarioJson,
+  scenarioString: string,
+  whichSteps: StepType,
+): TransLineProps[] {
+  const steps: TransLineProps[] = []
+  const json = scenarioObject[whichSteps]
+  for (const [index, value] of json.entries()) {
+    const keys = ['scenarios', scenarioString, whichSteps, index.toString()]
+    steps.push(buildTransLineProps(value, buildI18nKey(keys)))
   }
-  return transLineProps
+  return steps
 }
 
 /**
  * Get combined Claim Status content.
  */
 export default function getClaimStatus(scenarioType: ScenarioType): ClaimStatusContent {
-  getClaimStatusSummary(scenarioType)
-  const statusContent: ClaimStatusContent = {
-    heading: getClaimStatusHeading(scenarioType),
-    summary: getClaimStatusSummary(scenarioType),
-    nextSteps: [
-      'Ut enim ad minim veniam, quis nostrud exercitation ullamco laboris nisi ut aliquip ex ea commodo consequat.',
-      'Duis aute irure dolor in reprehenderit in voluptate velit esse cillum dolore eu fugiat nulla pariatur.',
-    ],
-  }
+  // Explicitly cast the scenario string (e.g. scenario1, scenario2) into the union of literal types
+  // expected by Typescript.
+  const scenarioString = scenarioToString(scenarioType) as keyof typeof claimStatusJson.scenarios
+
+  const scenarioObject = claimStatusJson.scenarios[scenarioString]
 
-  return statusContent
+  return {
+    heading: buildClaimStatusHeading(scenarioType),
+    summary: buildClaimStatusSummary(scenarioObject, scenarioString),
+    yourNextSteps: buildNextSteps(scenarioObject, scenarioString, 'your-next-steps'),
+    eddNextSteps: buildNextSteps(scenarioObject, scenarioString, 'edd-next-steps'),
+  }
```

Comment:
github mangled this file's diff so much that I had an excuse to use `vimdiff` for the first time in _ages_! 

Really like this `return` refactor. 👨🏼‍🍳👌🏼